Python/pythonrun.c

```diff
@@ -952,6 +946,118 @@ print_exception(struct exception_print_context *ctx, PyObject *value)
         err = _PyTraceBack_Print_Indented(
             tb, EXC_INDENT(ctx), EXC_MARGIN(ctx), header_margin, header, f);
     }
+    Py_XDECREF(tb);
+    return err;
+}
+
+/* Prints the message line: module.qualname[: str(exc)] */
+static int
+print_exception_message(struct exception_print_context *ctx,
+                        PyObject *type,
+                        PyObject *value)
+{
+    PyObject *f = ctx->file;
+
+    PyObject *modulename = NULL;
+    PyObject *qualname = NULL;
+    PyObject *s = NULL;
+
+    _Py_IDENTIFIER(__module__);
+
+    assert(PyExceptionClass_Check(type));
+
+    if (write_indented_margin(ctx, f) < 0) {
+        goto error;
+    }
+    modulename = _PyObject_GetAttrId(type, &PyId___module__);
+    if (modulename == NULL || !PyUnicode_Check(modulename)) {
+        Py_CLEAR(modulename);
+        PyErr_Clear();
+        if (PyFile_WriteString("<unknown>.", f) < 0) {
+            goto error;
+        }
+    }
+    else {
+        if (!_PyUnicode_EqualToASCIIId(modulename, &PyId_builtins) &&
+            !_PyUnicode_EqualToASCIIId(modulename, &PyId___main__))
+        {
+            if (PyFile_WriteObject(modulename, f, Py_PRINT_RAW)) {
+                goto error;
+            }
+            if (PyFile_WriteString(".", f) < 0) {
+                goto error;
+            }
+        }
+        Py_CLEAR(modulename);
+    }
+
+    qualname = PyType_GetQualName((PyTypeObject *)type);
+    if (qualname == NULL || !PyUnicode_Check(qualname)) {
+        Py_CLEAR(qualname);
+        PyErr_Clear();
+        if (PyFile_WriteString("<unknown>", f) < 0) {
+            goto error;
+        }
+    }
+    else {
+        if (PyFile_WriteObject(qualname, f, Py_PRINT_RAW) < 0) {
+            goto error;
+        }
+        Py_CLEAR(qualname);
+    }
+
+    if (value != Py_None) {
```

Comment:
```suggestion
    if (!Py_IsNone(value)) {
```

Actually, I'd reverse that check and `return 0`. The rest of the function then saves an indent level ⬅️ 🎉